### PR TITLE
CU/7414-Liz-disability-rating-downtime-fix

### PR DESCRIPTION
**Pull request by liztownd**
## Description of Change
Added a check for maintenance window downtime for disability rating and military service history to the Sync Screen for Veteran Status.

## Screenshots/Video
Veteran Status page with disability maintenance window in effect before/after:

&nbsp;&nbsp;

Verified that the maintenance window downtime was in effect as in each screenshot above the maintenance notice was visible on the Benefits/DisabilityRating screen:

When Military Service History is in downtime (Note: Veteran Status is disabled as the Service History is a requirement for that screen):

&nbsp;&nbsp;

## Testing
Also updated the errorSlices test to include a check for the mock disability rating maintenance window

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
When there is a downtime maintenance window for disability rating it should not call the API for disability rating.

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Comment by liztownd** [comment]
@TKDickson I did notice that there is not a maintenance downtime check/block for the military service history call in the Sync Screen as well, didn't know whether or not that needed aded here as well or not.

File: VAMobile/src/screens/SyncScreen/SyncScreen.tsx [hunk 1]
```diff
@@ -31,6 +31,9 @@ const SyncScreen: FC<SyncScreenProps> = () => {
   const { preloadComplete: militaryHistoryLoaded, loading: militaryHistoryLoading } = useSelector<RootState, MilitaryServiceState>((s) => s.militaryService)
   const { preloadComplete: disabilityRatingLoaded, loading: disabilityRatingLoading } = useSelector<RootState, DisabilityRatingState>((s) => s.disabilityRating)
   const { data: userAuthorizedServices, isLoading: loadingUserAuthorizedServices } = useAuthorizedServices({ enabled: loggedIn })
+  // TODO: For some reason Unit Tests cannot pick up the DowntimeFeatureTypeConstants constant
```
Comment (liztownd) [inline VAMobile/src/screens/SyncScreen/SyncScreen.tsx:34]: I do not know why, but for whatever reason when the constants are used here the Unit Tests fail. I have spent almost 10 hours on Unit Tests alone trying to make it work. @Sparowhawk has also looked at the Unit Test and confirmed that there is something wonky going on with the tests. Eventually, yes, let's move back to the `DowntimeFeatureTypeConstants` usage (hence the `TODO` tag).

Comment (Sparowhawk) [reply VAMobile/src/screens/SyncScreen/SyncScreen.tsx:34]: can you create a CU ticket for that?

Comment (liztownd) [reply VAMobile/src/screens/SyncScreen/SyncScreen.tsx:34]: Done. [GH Issue #7462](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/7462)

**Review by Sparowhawk — APPROVED** [review]

**Review by dumathane — APPROVED** [review]

**Review by rbontrager — APPROVED** [review]

**Pull request merged**